pandas/core/arrays/base.py

```diff
@@ -610,3 +616,122 @@ def _ndarray_values(self):
         used for interacting with our indexers.
         """
         return np.array(self)
+
+
+class ExtensionScalarOpsMixin(object):
+    """
+    A base class for the mixins for different operators.
+    Can also be used to define an individual method for a specific
+    operator using the class method create_method()
+    """
+
+    @classmethod
+    def _create_method(cls, op, coerce_to_dtype=True):
+        """
+        A class method that returns a method that will correspond to an
+        operator for an ExtensionArray subclass, by dispatching to the
+        relevant operator defined on the individual elements of the
+        ExtensionArray.
+
+        Parameters
+        ----------
+        op: function
+            An operator that takes arguments op(a, b)
+        coerce_to_dtype:  bool
+            boolean indicating whether to attempt to convert
+            the result to the underlying ExtensionArray dtype
+            (default True)
+
+        Returns
+        -------
+        A method that can be bound to a method of a class
+
+        Example
+        -------
+        Given an ExtensionArray subclass called MyClass, use
+
+        >>> __add__ = ExtensionScalarOpsMixin.create_method(operator.add)
+
+        in the class definition of MyClass to create the operator
+        for addition.
+
+        """
+
+        def _binop(self, other):
+            def convert_values(param):
+                if isinstance(param, ExtensionArray) or is_list_like(param):
+                    ovalues = param
+                else:  # Assume its an object
+                    ovalues = [param] * len(self)
+                return ovalues
+            lvalues = self
+            rvalues = convert_values(other)
+
+            # If the operator is not defined for the underlying objects,
+            # a TypeError should be raised
+            res = [op(a, b) for (a, b) in zip(lvalues, rvalues)]
+
+            if coerce_to_dtype:
+                try:
+                    res = self._from_sequence(res)
+                except TypeError:
+                    pass
+
+            return res
+
+        op_name = _get_op_name(op, True)
+        return set_function_name(_binop, op_name, cls)
+
+
+class ExtensionScalarArithmeticMixin(ExtensionScalarOpsMixin):
```

Comment:
still not clear to why this inherits ``ExtensionScalarOpsMixin``. If I want *only* the ops definitions, and want to define my own create_method this seems a bit odd to do. Rather I would like to have a ``_create_method`` that just raises NotImplementedError, so things fail as soon as I inherit this base class. 

the way this is written it is entirely too error prone. You drop in these classes, then it *might* work incorrectly! e.g. maybe I happen to have a scalar ops defined. This should be very explict. I dont' see ANY advantage of defaulting to a scalar ops schema here.

Reply:
@jreback I think I see what you are saying, so I have pushed a commit that I think does what you suggest.  I've created infrastructure that allows a developer to define their own `create_method`. I've written up docs that explains how it works.  The underlying implementation is a little complex, because there are lots of inner classes flying around, but now people can use the `ExtensionScalarArithmeticMixin` if they have the underlying objects with operators defined, or they can use `ExtensionArithmeticMixin(MyMixin)` to define all of the operators, and then the class `MyMixin` has the definition of `create_method` in it.

I hope this addresses your concern.

Reply:
Personally, I find this much more complicated than what you had before, and I am also not really sure if it will be that useful in practice (i.e. the usecase of wanting to implement your own `_create_method` but that works for all operators. Which actually was also possible with your previous approach by inheriting eg ExtensionScalarArithmeticMixin and overriding the `_create_method` method). 
I would rather wait and see if there is need for supporting this usecase.

Reply:
> I dont' see ANY advantage of defaulting to a scalar ops schema here.

@jreback again, 1) it's *not* defaulting for the generic ExtensionArray, the scalar ops is only implemented for this *optional* mixin 2) this is entirely the point of this PR. 
Yes, you may well never want to use such fallback to the scalars. But then you just don't use those mixins from this PR.

This PR is not intending to solve the general ops issue for ExtensionArrays, its purpose was to provide a easy solution for a very specific use case.